src/training_pipeline.py

```diff
@@ -64,10 +65,18 @@ def train(config: DictConfig) -> Optional[float]:
                 logger.append(hydra.utils.instantiate(lg_conf))
 
     # Init lightning trainer
+    if config.trainer.get("amp_backend"):
+        log.info(f"Using Mixed Precision <{config.trainer.amp_backend}>")
+    if config.trainer.get("precision"):
+        log.info(f"Using precision training <{config.trainer.precision}>")
     log.info(f"Instantiating trainer <{config.trainer._target_}>")
     trainer: Trainer = hydra.utils.instantiate(
         config.trainer, callbacks=callbacks, logger=logger, _convert_="partial"
     )
+    
+    # print model GPU
+    for i in range(torch.cuda.device_count()):
+        print(f"\n> GPU {i} ready: {torch.cuda.get_device_name(i)}")
```

Comment:
Is there a reason for printing the available cuda devices in the middle of the training pipeline? The information about number of used GPUs is already printed by the lightning the beginning of the training:
![image](https://user-images.githubusercontent.com/43458894/156948789-fafb2262-a218-4e8e-8bc2-9c044b2bd2a5.png)